server/src/models/User.ts

```diff
@@ -0,0 +1,37 @@
+import { ObjectId, Schema, Model, model } from "mongoose";
+
+export interface UserAttributes {
+  saved_restaurants: ObjectId;
+}
+
+export interface UserDocument extends UserAttributes {
+  _id: ObjectId;
+  auth0_id: string;
+  created_at: Date;
+}
```

Comment:
no need to extend these guys, just put them together as the UserAttributes (What does a user document have?)

something like:
UserAttributes{
id: ObjectId,
auth0_id: ...,
saved_restaurants: ...,
created_at: ...
}